### PR TITLE
refactor state and storage behind one interface

### LINKED ISSUE #5329: refactor StateProvider and StorageProvider behind one interface IWorldState
With Verkle Trees, we dont have seperate state and storage tree and there is no concept of storage root. Everything is stored in a single tree

### PR DESCRIPTION
**Pull request by tanishqjasoria**
Resolves #5329 
Replaces #5533 

some interface and API changes for integrating verkle trees with nethermind. 
Also, bring state access under limited interfaces across the solution.

## Changes

- use WorldState instead of State and Storage provider
- merge the functions UpdateCode and UpdateCodeHash  in StateProvider

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a fix or a feature that would cause existing functionality not to work as expected)
- [ ] Documentation update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional or API changes)
- [ ] Build-related changes
- [x] Other: _description_ 

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [x] Yes
- [] No

#### Notes on testing
synced a node using snap sync and let it run for a few hours after syncing - everything looked normal

### CODE REVIEW

**Comment by tanishqjasoria** [comment]
@LukaszRozmej @Scooletz now worldState delegates to internal classes StateProvider PersistentStorageProvider TransientStorageProvider
@asdacap the transient and persistent storage is handled by different classes

File: src/Nethermind/Nethermind.Benchmark/State/StorageCellBenchmark.cs [hunk 1]
```diff
@@ -7,6 +7,7 @@
 using Nethermind.Core;
 using Nethermind.Int256;
 using Nethermind.State;
+using Nethermind.State.Tracing;
```
Comment (asdacap) [inline src/Nethermind/Nethermind.Benchmark/State/StorageCellBenchmark.cs:10]: NP: Accidental?

File: src/Nethermind/Nethermind.PerfTest/Program.cs [hunk 2]
```diff
@@ -332,7 +332,7 @@ private static async Task RunBenchmarkBlocks()
                 if (allocation.Code != null)
                 {
                     Keccak codeHash = stateProvider.UpdateCode(allocation.Code);
-                    stateProvider.UpdateCodeHash(address, codeHash, specProvider.GenesisSpec);
+                    stateProvider.InsertCode(address, codeHash, specProvider.GenesisSpec);
```
Comment (asdacap) [inline src/Nethermind/Nethermind.PerfTest/Program.cs:335]: Isn't the second param the code itself?

**Review by asdacap — APPROVED** [review]
Have you tested on mainnet?

File: src/Nethermind/Nethermind.Benchmark/State/StorageCellBenchmark.cs (hunk 1 above)
Comment (tanishqjasoria) [reply src/Nethermind/Nethermind.Benchmark/State/StorageCellBenchmark.cs:10]: moved tracers to Nethermind.State.Tracing - that is why the additional import

File: src/Nethermind/Nethermind.PerfTest/Program.cs (hunk 2 above)
Comment (tanishqjasoria) [reply src/Nethermind/Nethermind.PerfTest/Program.cs:335]: ahh, yes - my bad

Comment (tanishqjasoria) [reply src/Nethermind/Nethermind.PerfTest/Program.cs]: which solution is this a part of? i cant find the file

**Comment by tanishqjasoria** [comment]
> Have you tested on mainnet?

yes, works fine!

**Review by LukaszRozmej — APPROVED** [review]

**Pull request merged**